### PR TITLE
DOCSP-34836 Playgrounds Default DB in Connection String

### PR DESCRIPTION
**Pull request by ajhuh-mdb**
## DESCRIPTION
- Updated steps to use procedure directive
- Included note on behavior when connection string specifies a default database and how to use a a different one if necessary.

## STAGING
https://preview-mongodbajhuhmdb.gatsbyjs.io/mongodb-vscode/DOCSP-34836-playgrounds-database-connection-string/playgrounds/#connect-to-a-deployment-from-a-playground

## JIRA
https://jira.mongodb.org/browse/DOCSP-34836

## BUILD LOG
https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=65738aa37a5cb5fef5487d5a

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Is this free of spelling errors?
- [ ] Is this free of grammatical errors?
- [ ] Is this free of staging / rendering issues?
- [ ] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

File: source/playgrounds.txt [hunk 1]
```diff
@@ -150,19 +150,41 @@ If you have a playground file open in |vscode-short| and do not have
...
+.. procedure::
+   :style: normal
    
-#. Press :guilabel:`Enter`.
-
-#. Run your playground.
+   .. step:: Select :guilabel:`Click here to add connection` 
+
+      After you click :guilabel:`Click here to add connection`, |vsce| opens the 
+      connection string dialog.
```
Comment (ianf-mongodb) [inline source/playgrounds.txt:159]:
```suggestion
      After you click :guilabel:`Click here to add connection`, the
      connection string dialog displays.
```

File: source/playgrounds.txt [hunk 2]
```diff
@@ -150,19 +150,41 @@ If you have a playground file open in |vscode-short| and do not have
...
+
+      .. figure:: /images/vsce-playground-add-connection-btn.png
+         :figwidth: 700px
+         :alt: Link to add connection from playground
+         
+   .. step:: Enter your connection string
+      
+      Enter the :manual:`connection string </reference/connection-string/>` for 
+      the deployment you want to run this playground against.
```
Comment (ianf-mongodb) [inline source/playgrounds.txt:168]:
```suggestion
      Enter a :manual:`connection string </reference/connection-string/>` to connect to your deployment.
```

File: source/playgrounds.txt [hunk 3]
```diff
@@ -150,19 +150,41 @@ If you have a playground file open in |vscode-short| and do not have
...
+
+      .. note:: 
+         
+         If the connection string specifies a database, the playground runs
```
Comment (ianf-mongodb) [inline source/playgrounds.txt:172]: **suggestion**

Should this be does not?

```suggestion
         If the connection string does not specify a database, the playground runs 
```

File: source/playgrounds.txt [hunk 4]
```diff
@@ -150,19 +150,41 @@ If you have a playground file open in |vscode-short| and do not have
...
+   .. step:: Enter your connection string
+      
+      Enter the :manual:`connection string </reference/connection-string/>` for 
+      the deployment you want to run this playground against.
+
+      .. note:: 
+         
+         If the connection string specifies a database, the playground runs 
+         against the default connected database unless you explicitly call 
+         ``use('<database_name>')``  to use another database.
```
Comment (ianf-mongodb) [inline source/playgrounds.txt:174]:
```suggestion
         against the default connected database unless you explicitly switch databases with 
         ``use('<database_name>')``.
```

File: source/playgrounds.txt [hunk 5]
```diff
@@ -150,19 +150,41 @@ If you have a playground file open in |vscode-short| and do not have
...
+
+         If you are connected to a default database, the playground 
+         autocompletes only for collection names available on the default 
+         database.
```
Comment (ianf-mongodb) [inline source/playgrounds.txt:178]:
```suggestion
         autocompletes only for collection names available on that database. 
```

File: source/playgrounds.txt [hunk 6]
```diff
@@ -150,19 +150,41 @@ If you have a playground file open in |vscode-short| and do not have
...
+   .. step:: Select :guilabel:`Click here to add connection` 
+
+      After you click :guilabel:`Click here to add connection`, |vsce| opens the 
+      connection string dialog.
+
+      .. figure:: /images/vsce-playground-add-connection-btn.png
+         :figwidth: 700px
+         :alt: Link to add connection from playground
+         
+   .. step:: Enter your connection string
```
Comment (ianf-mongodb) [inline source/playgrounds.txt:165]: I had to click on extra time before the connection string dialog appeared:

Screenshot 2023-12-08 at 5 34 38 PM

File: source/playgrounds.txt [hunk 7]
```diff
@@ -150,19 +150,41 @@ If you have a playground file open in |vscode-short| and do not have
...
+      
+      Enter the :manual:`connection string </reference/connection-string/>` for
```
Comment (ianf-mongodb) [inline source/playgrounds.txt:167]: **Suggestion**

It may be valuable to enter a small sample connection string here to avoid the users needing to go to the additional page WDYT?
Something like:

```
mongodb://localhost:27017/mydatabase
```

File: source/playgrounds.txt [hunk 8]
```diff
@@ -150,19 +150,41 @@ If you have a playground file open in |vscode-short| and do not have
...
+         If you are connected to a default database, the playground 
+         autocompletes only for collection names available on the default 
+         database. 
+         
+   .. step:: Press :guilabel:`Enter`
+
+      After you enter your connection string, a confirmation message replaces 
+      the :guilabel:`Click here to add connection` link. The confirmation 
+      message includes information on the connected deployment and, if 
+      applicable, the default database.
```
Comment (ianf-mongodb) [inline source/playgrounds.txt:185]: I think you can shorten this to:

The confirmation message includes information on the connected deployment.

I dont see anything siginicant on the default database information, let me know if I am looking at the wrong spot!:

 
Screenshot 2023-12-08 at 5 40 08 PM

**Review by ianf-mongodb — COMMENTED** [review]
Hi @ajhuh-mdb thanks for picking this up, I left a few suggestions and edits for the PR.

File: source/playgrounds.txt (hunk 1 above)
Comment (ajhuh-mdb) [reply source/playgrounds.txt:159]: Opting to find a middle ground here since ending the sentence with "display" makes it sound like there's a missing a direct object.

File: source/playgrounds.txt (hunk 3 above)
Comment (ajhuh-mdb) [reply source/playgrounds.txt:172]: Ah I might've worded it weirdly. the behavior is tht whatever database is specified in the connection string **is** the default database.

File: source/playgrounds.txt (hunk 4 above)
Comment (ajhuh-mdb) [reply source/playgrounds.txt:174]: Split this into two sentences. LMK what you think.

File: source/playgrounds.txt (hunk 6 above)
Comment (ajhuh-mdb) [reply source/playgrounds.txt:165]: Added an additional step about this + included information on how previously connected deployments will show up as options.

File: source/playgrounds.txt (hunk 8 above)
Comment (ajhuh-mdb) [reply source/playgrounds.txt:185]: The original ticket filer wanted us to make a note about this because multiple users have shared feedback that weren't aware they were in the incorrect database, which prevented them from connecting to collections in other dbs. 

connection-with-default-connected-db

Comment (ajhuh-mdb) [reply source/playgrounds.txt:185]: I can change the wording to say "current" database rather than default.

File: source/playgrounds.txt (hunk 7 above)
Comment (ajhuh-mdb) [reply source/playgrounds.txt:167]: I'm leaning towards keeping this as is because I'm expecting that a user coming to use the VSCode extension playground would be familiar with MDB basics such as using a connection string, given they must already have a deployment set up.

**Comment by ajhuh-mdb** [comment]
Thanks for the thorough review @ianf-mongodb! I just finished going through your comments and put in my edits/responses. LMK what you think and if you have any follow-up questions.

**Review by ianf-mongodb — APPROVED** [review]
Thanks for the updates and the feedback Ali, this one LGTM!

File: source/playgrounds.txt [hunk 9]
```diff
@@ -150,19 +150,48 @@ If you have a playground file open in |vscode-short| and do not have
...
+         against that database by default. To switch databases,   
+         call ``use('<database_name>')``.
+
+         If you are connected to a default database, the playground 
+         autocompletes only for collection names available on that 
+         database. 
+         
+   .. step:: Press :guilabel:`Enter`
+
+      After you enter your connection string, a confirmation message replaces
```
Comment (Anemy) [inline source/playgrounds.txt:189]: Nit, fine as is, in VSCode these `confirmation` messages are called `code lenses`. https://code.visualstudio.com/blogs/2017/02/12/code-lens-roundup

**Review by Anemy — APPROVED** [review]

File: source/playgrounds.txt (hunk 9 above)
Comment (ajhuh-mdb) [reply source/playgrounds.txt:189]: Gotcha, thanks! Will update.
